### PR TITLE
chore: replace shortened URL with full URL

### PR DESCRIPTION
**Pull request by tim-finnigan**
*Description of changes:* Replace shortened URL with full URL. We don't want to link to external third-party link modifiers that might be changed or discontinued.

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jplock** [comment]
This was intentional and is using the shorten developed by the https://serverlessland.com team so we can track the number of clicks (460 as of today).

File: test/index.test.ts [hunk 1]
```diff
@@ -879,7 +879,7 @@ describe('Configure AWS Credentials', () => {
 
     expect(core.warning).toHaveBeenCalledWith(
       'To avoid using long-term AWS credentials, please update your workflows to authenticate using OpenID Connect.' +
-        ' See https://s12d.com/gha-oidc-aws for more information.'
+        ' See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services for more information.'
```
Comment (peterwoodworth) [inline test/index.test.ts:882]: Could link to the readme of this repo instead as well https://github.com/aws-actions/configure-aws-credentials#OIDC

**Comment by jplock** [comment]
The s12d.com domain and shortener service are managed by @singledigit. I think the click tracking is helpful, but if you want to change out the URL, I'm fine with whatever. I'm more interested in ensuring the warning message stays there (along with the fixes in #904)  👍

**Comment by tim-finnigan** [comment]
Thanks for providing that context. @kellertk asked me to create this PR, because he said we should link to the full URL so that we're not sending users to a domain that we don't control.

**Review by kellertk — APPROVED** [review]

**Pull request merged**